lib/ex_unit/lib/ex_unit/doc_test.ex

```diff
@@ -588,6 +601,15 @@ defmodule ExUnit.DocTest do
     end
   end
 
+  defp error_message_matches?(actual, expected) when actual == expected, do: true
+
+  defp error_message_matches?(actual, expected) do
+    case String.replace_suffix(expected, "...", "") do
+      ^expected -> false
```

Comment:
This seems to be an expensive check to verify if the string has ... or not. My suggestion is to use String.ends_with? and then `binary_slice`.